api/openapi-spec/openapi.yaml

```diff
@@ -0,0 +1,219 @@
+paths:
+# TODO GET endpoint for admin tasks - How to authenticate admin?
+# List all transactions with specific parameters
+  /v1/transactions/{debitor_id}:
+    get:
+      tags:
+        - transactions
+      summary: Request information about existing Transactions
+      description: Get existing transactions for an authorized user
+      operationId: getTransactionsByUser
+      parameters:
+        - name: debitor_id
+          in: path
+          description: The of a debitor.
+          required: true
+          schema:
+            type: integer
+            format: int64
+      responses:
+        '200':
+          description: Successful operation
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/TransactionResponse'
+        '400':
+          description: Invalid ID supplied
+        '401':
+          description: Request was unauthorized
+        '404':
+          description: No Transactions available for this attendee
+        '500':
+          description: An unexpected error occurred
+      security:
+        - api_key: []
+  /v1/transactions:
+    post:
+      tags:
+        - transactions
+      summary: Requests a new payment for a selected payment provider
+      description: |- 
+        Request provides information about a new payment that will be initiated based on the requested
+        payment method. 
+      operationId: createTransaction 
+      requestBody:
+        description: A transaction object to perform an operation according to the data provided
+        required: true
+        content:
+          application/json:
+            schema: 
+              $ref: '#/components/schemas/Transaction'
+      responses:
+        '200':
+          description: successful operation
+        '401':
+          description: Request was unauthorized
+        '405':
+          description: Request validation failed
+        '500':
+          description: An unexpected error occurred
+      security:
+        - api_key: [] 
+    put:
+      tags:
+        - "transactions"
+      summary: Updates transactions
+      description: |-
+        Updates transactions based on predefined rules. 
+      operationId: updateTransactions
+      requestBody:
+        required: true
+        description: A list of transactions that should be updated
+        content: 
+          application/json:
+            schema: 
+              $ref: '#/components/schemas/Transactions'
+      responses:
+        '200':
+          description: successful operation
+        '401':
+          description: Request was unauthorized
+        '405':
+          description: Request validation failed
+        '500':
+          description: An unexpected error occurred
+                
+              
+
+      security:
+        - api_key: []
+      
+components:
+  schemas:
+    TransactionResponse:
+      type: object
+      properties:
+        payload:
+          type: object
```

Comment:
"type: object" is not valid here according to OpenApi schema. Line should be removed.